addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/discovery/LGWebOSDiscovery.java

```diff
@@ -0,0 +1,221 @@
+        discoveryManager.start();
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        discoveryManager.stop();
+    }
+
+    // DiscoveryManagerListener
+
+    @Override
+    public void onDeviceAdded(DiscoveryManager manager, ConnectableDevice device) {
+        thingDiscovered(createDiscoveryResult(device));
+    }
+
+    @Override
+    public void onDeviceUpdated(DiscoveryManager manager, ConnectableDevice device) {
+        logger.debug("Device updated: {}", device);
+        thingRemoved(createThingUID(device));
+        thingDiscovered(createDiscoveryResult(device));
+    }
+
+    @Override
+    public void onDeviceRemoved(DiscoveryManager manager, ConnectableDevice device) {
+        logger.debug("Device removed: {}", device);
+        thingRemoved(createThingUID(device));
+    }
+
+    @Override
+    public void onDiscoveryFailed(DiscoveryManager manager, ServiceCommandError error) {
+        logger.warn("Discovery Failed {}", error.getMessage());
+    }
+
+    // Helpers for DiscoveryManagerListener Impl
+    private DiscoveryResult createDiscoveryResult(ConnectableDevice device) {
+        ThingUID thingUID = createThingUID(device);
+        DiscoveryResult result = DiscoveryResultBuilder.create(thingUID)
+                .withProperty(PROPERTY_IP_ADDRESS, device.getIpAddress()).withLabel(device.getFriendlyName()).build();
+        return result;
+    }
+
+    private ThingUID createThingUID(ConnectableDevice device) {
+        return new ThingUID(THING_TYPE_WEBOSTV, device.getIpAddress().replace('.', '_'));
+    }
+
+    public DiscoveryManager getDiscoveryManager() {
+        return this.discoveryManager;
+    }
+
+    // Context for connect sdk
+    private final String DATA_DIR = new File("etc" + File.separator + "connect_sdk").getAbsolutePath();
+
+    @Override
+    public String getDataDir() {
+        return DATA_DIR;
+    }
+
+    @Override
+    public String getApplicationName() {
+        return "Openhab";
+    }
+
+    @Override
+    public String getPackageName() {
+        return "org.openhab";
+    }
+
+    @Override
+    public InetAddress getIpAddress() {
+        return localInetAddresses;
+    }
+
+    /**
+     * Get local IP either through configuration or auto detection.
+     * Method will ignore loopback addresses.
+     *
+     * @param localIP optional configuration string
+     * @return local ip or <code>null</code> if detection was not possible.
+     */
+    private InetAddress findLocalInetAddresses(String localIP) {
+        // evaluate optional localIP parameter, can be configured through config admin (lgwebos.cfg)
+        if (localIP != null && !localIP.trim().isEmpty()) {
+            try {
+                logger.debug("localIP parameter explicitly set to: {}", localIP);
+                return InetAddress.getByName(localIP.trim());
+            } catch (UnknownHostException e) {
+                logger.warn("localIP config parameter could not be parsed: {}", localIP);
```

Comment:
If you show exception's getMessage(), there will be detailed explanation what's wrong with the localIP value.

Reply:
will add